### DISCUSSION TITLE
How to run multiple prompts back to back?

### DISCUSSION THREAD

**Discussion by Theagainmen** [hlky/stable-diffusion-webui#618] [General]
Hi all,

Thanks for this great repo!

When I have to leave my computer I would like to insert multiple prompts that run back to back, given the processing time this would be very handy.

I've enabled in the advanced settings "Create prompt matrix" but I've noticed that this does not do what I thought it would. I'm wondering is there a way that does the following:

I insert:
_Prompt 1
Prompt 2
Prompt 3_

**When I hit generate each prompt will be rendered back to back, not a _combination_ of prompt 1 & 3 for example, just _each exact_ prompt.** 

Thanks for any help!

**Comment by athu16** [comment]
You can enable the prompt matrix, then type
`@(Prompt 1 | Prompt 2 | Prompt 3)`
That's the easiest way to do it as of now
If you have "save individual images" enabled, each prompt will be saved in its own folder

**Reply by Theagainmen** [reply]
Thank you so much, works exactly as I wanted!! What does the @ exactly mean in this context?

**Reply by BilMorJr** [reply]
Putting @ at the start of a prompt causes it to evaluate any parenthesis with |'s as exclusive lists.  If you don't put the @ at the start of the prompt, it will treat |'s as non-exclusive. So:

@A (cat|dog|mouse)
will run "A cat", "A dog", and "A mouse"

A cat|dog|mouse
will run
"A cat"
"A cat, dog"
"A cat, mouse"
"A cat, mouse, dog"

**Reply by Theagainmen** [reply]
Amazing explanation, thanks a lot!
